### ISSUE TITLE
Release a tokio-postgres alpha with async/await support

### ISSUE THREAD

**Issue by djc** [sfackler/rust-postgres#493]
Would be nice if only for having docs on docs.rs, and a release point.

**Comment by sfackler** [comment]
Seems reasonable! There are some in-flight changes I'm working on right now to the query methods, but once those land I'll cut an alpha release.

**Comment by djc** [comment]
Would that include building on `TryStream`? I'm just trying to bootstrap my use of tokio-postgres in a greenfield app that I'm building with async/await, but so far getting the first `query()` use past the compiler is proving difficult. Might also need to add some `+ Unpin` bounds?

**Comment by sfackler** [comment]
If you want to manually iterate through a stream with `.next()`, you need to pin it (via e.g. `Box::pin` or the pin_utils crate). That's just how streams work though.

One of the changes I am planning on making is having `query()` just return a `Vec<Row>`, with `query_raw` being the "maximally flexible" version that returns a stream.

**Comment by djc** [comment]
It might also be nice to have a `query_row()` like rusqlite:

https://docs.rs/rusqlite/0.20.0/rusqlite/struct.Connection.html#method.query_row

**Comment by djc** [comment]
`query_row()` is great, I was able to remove the use of `pin_mut!()` again for now. Do you have some kind of ETA/open issue list for things that are blocking an alpha or beta release?

**Comment by sfackler** [comment]
I think everything should be just about ready to go. I'll make one last check over the API tonight and cut a release.

**Comment by djc** [comment]
Awesome, thanks!

**Comment by sfackler** [comment]
Published!

**Comment by importcjj** [comment]
There is an asynchronous connection pool with async/await support. https://github.com/importcjj/mobc
